chainer/optimizer.py

```diff
@@ -853,3 +853,58 @@ def __call__(self, rule, param):
         xp = cuda.get_array_module(grad)
         with cuda.get_device_from_array(grad):
             xp.clip(grad, self.lower_bound, self.upper_bound, out=grad)
+
+
+class GradientLARS(object):
+
+    """Optimizer/UpdateRule hook function for layer wise adaptive rate scaling.
+
+    See: `Large Batch Training of Convolutional Networks \
+          <https://arxiv.org/abs/1708.03888>`_
+
+    This hook function scales all gradient arrays to fit to the weight norm.
+
+    Args:
+        threashold (float): If weight norm is more than threshold,
+            this function scales all gradient arrays to fit weight norm.
+        weight_decay (float): Coefficient for the weight decay.
+        eps (float): Small value for the numerical stability.
+
+    Attributes:
+        threashold (float): If weight norm is more than threshold,
+            this function scales all gradient arrays to fit weight norm.
+        weight_decay (float): Coefficient for the weight decay.
+        eps (float): Small value for the numerical stability.
+    """
+    name = 'GradientLARS'
+    call_for_each_param = True
+
+    def __init__(self, threshold=1e-2, weight_decay=0.0, eps=1e-9):
+        self.threshold = threshold
+        self.weight_decay = weight_decay
+        self.eps = eps
+
+    def __call__(self, rule, param):
+        p, g = param.data, param.grad
+        if p is None or g is None:
+            return
+
+        xp = cuda.get_array_module(p)
+
+        # weight norm
+        p_norm = xp.linalg.norm(p)
+        # grad norm
+        g_norm = xp.linalg.norm(g)
+        local_rate = p_norm / (self.eps + g_norm + self.weight_decay * p_norm)
```

Comment:
It could be confusing to call this value `local_rate` because local LR (learning rate) λ^l is defined as multiplication of η (LARS coefficient) and this value in (6) of https://arxiv.org/pdf/1708.03888.pdf.

Reply:
OK.